### PR TITLE
feat: Add version string to build artifacts

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

Related to https://github.com/noir-lang/noir/issues/3130

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by TomAFrench** [comment]
> To get there, we need to tag each json file with an extra field "noir_version" and possibly "acvm_version". When we compile, we can check if the artifact is already there and then check the noir_version. If the noir_version has changed, then we recompile.

This won't work as we'll fail inside serde before we check the versions. We should just remove the unwraps when deserializing circuits and then if it returns an error, we disregard the build artifact.

**Comment by kevaundray** [comment]
> This won't work as we'll fail inside serde before we check the versions. We sh

Good catch -- This PR would only deal with changes in Noir that do not modify serialization

**Comment by kevaundray** [comment]
> > To get there, we need to tag each json file with an extra field "noir_version" and possibly "acvm_version". When we compile, we can check if the artifact is already there and then check the noir_version. If the noir_version has changed, then we recompile.
> 
> This won't work as we'll fail inside serde before we check the versions. We should just remove the unwraps when deserializing circuits and then if it returns an error, we disregard the build artifact.

modified it to return an error -- if there is an error, then there is no CachedProgram and the program should recompile -- I'll do some testing by installing nargo with the version before your serialization changes and then compile with this version with the old artifacts

File: compiler/wasm/src/compile.rs [hunk 1]
```diff
@@ -17,6 +17,9 @@ use crate::errors::JsCompileError;
 
 const BACKEND_IDENTIFIER: &str = "acvm-backend-barretenberg";
 
+const NOIR_ARTIFACT_VERSION_STRING: &str =
+    concat!(env!("CARGO_PKG_VERSION"), "-", env!("GIT_COMMIT"));
+
```
Comment (kevaundray) [inline compiler/wasm/src/compile.rs:22]: These two being here seems a bit odd -- I think versioning could probably go into the driver/nargo and then have it be shared between nargo_cli and wasm

**Comment by TomAFrench** [comment]
Superceded by #3248

**Pull request closed without merging**